utils/build-script-impl

```diff
@@ -115,6 +115,14 @@ KNOWN_SETTINGS=(
     darwin-toolchain-version                      ""                "Version for xctoolchain info plist and installer pkg"
     darwin-xcrun-toolchain                        "default"         "the name of the toolchain to use on Darwin"
 
+    ## WebAssembly/WASI Options
+    wasi-sysroot                                  ""                "An absolute path to the WASI sysroot that will be used as a libc implementation for WebAssembly builds"
+    wasi-icu-uc                                   ""                "Path to libicuuc.a"
+    wasi-icu-uc-include                           ""                "Path to a directory containing headers for libicuuc"
+    wasi-icu-i18n                                 ""                "Path to libicui18n.a"
+    wasi-icu-i18n-include                         ""                "Path to a directory containing headers libicui18n"
+    wasi-icu-data                                 ""                "Path to libicudata.a"
+
```

Comment:
I really would rather that we hold off on this, in currently restructuring the build and most of this is just going to make it harder to transform.

Reply:
Would be happy to help in cleaning this up and to avoid conflicts, just need to understand what the long-term vision here is and how we should coordinate work in this area. Could you link to some of the related changes or maybe write ups on how this is supposed to work in your opinion?

Reply:
@kateinoigakukun there seems to be a plan in the works to eventually eliminate `build-script-impl`. I wonder if it would be best to remove the changes to this file from the PR and we can land them later, probably in a different form as a separate PR, after @compnerd has finished his changes to the build system.